java/code/src/com/redhat/rhn/domain/channel/ChannelFactory.java

```diff
@@ -1366,12 +1366,24 @@ public static List<Channel> listCustomChannels() {
                 .getResultList();
     }
 
+    /**
+     * List all vendor channels (org is null) that should be synchronized automatically
+     * @return list of vendor channels
+     */
+    public static List<Channel> listVendorChannelsForAutoSync() {
+        List<Channel> result = singleton.listObjectsByNamedQuery("Channel.findVendorChannelsForAutoSync", Map.of());
+        if (result != null) {
+            return result;
+        }
+        return new ArrayList<>();
```

Comment:
I know this was originally my code 🤦‍♂️ , but I would invert this if to make more readable for the correct flow and use List.of() so we don't create a new list instance:

```java
        if (result == null) {
            return List.of();
        }

        return result;
```

Same for `listCustomChannelsWithRepositoriesForAutoSync()` 🙂